tests/cloner_test.go

```diff
@@ -2537,6 +2538,106 @@ var _ = Describe("all clone tests", func() {
+			Expect(err).ToNot(HaveOccurred())
+
+			snapClass := f.GetSnapshotClass()
+			snapshot = &snapshotv1.VolumeSnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "snap-" + snapSourceDv.Name,
+					Namespace: f.Namespace.Name,
+				},
+				Spec: snapshotv1.VolumeSnapshotSpec{
+					Source: snapshotv1.VolumeSnapshotSource{
+						PersistentVolumeClaimName: &pvc.Name,
+					},
+					VolumeSnapshotClassName: &snapClass.Name,
+				},
+			}
+			err = f.CrClient.Create(context.TODO(), snapshot)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() bool {
+				err = f.CrClient.Get(context.TODO(), crclient.ObjectKeyFromObject(snapshot), snapshot)
+				if err != nil {
+					return false
+				}
+				return snapshot.Status != nil && snapshot.Status.ReadyToUse != nil && *snapshot.Status.ReadyToUse
+			}, 10*time.Second, 1*time.Second).Should(BeTrue())
+			By("Snapshot ready, no need to keep PVC around")
+			err = f.DeletePVC(pvc)
+			Expect(err).ToNot(HaveOccurred())
+
+			for i = 0; i < repeat; i++ {
+				dataVolume := utils.NewDataVolumeForCloningFromSnapshot(fmt.Sprintf("clone-from-snap-%d", i), size, snapshot.Namespace, snapshot.Name, nil, &volumeMode)
+				By(fmt.Sprintf("Create new datavolume %s which will clone from volumesnapshot", dataVolume.Name))
+				dataVolume, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolume)
+				Expect(err).ToNot(HaveOccurred())
+				f.ForceBindPvcIfDvIsWaitForFirstConsumer(dataVolume)
+			}
+
+			for i = 0; i < repeat; i++ {
+				By("Waiting for clones to be completed")
+				dvName := fmt.Sprintf("clone-from-snap-%d", i)
+				err = utils.WaitForDataVolumePhase(f, f.Namespace.Name, cdiv1.Succeeded, dvName)
+				Expect(err).ToNot(HaveOccurred())
+				_, err = f.K8sClient.CoreV1().PersistentVolumeClaims(f.Namespace.Name).Get(context.TODO(), dvName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			By("Verify MD5 on one of the DVs")
+			lastDvName := fmt.Sprintf("clone-from-snap-%d", i-1)
+			pvc, err = f.K8sClient.CoreV1().PersistentVolumeClaims(f.Namespace.Name).Get(context.TODO(), lastDvName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			path := utils.DefaultImagePath
+			if volumeMode == v1.PersistentVolumeBlock {
+				path = utils.DefaultPvcMountPath
+			}
+			same, err := f.VerifyTargetPVCContentMD5(f.Namespace, pvc, path, utils.UploadFileMD5, utils.UploadFileSize)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(same).To(BeTrue())
+		},
+			Entry("with filesystem single clone", v1.PersistentVolumeMode(v1.PersistentVolumeFilesystem), 1),
+			Entry("with filesystem multiple clones", v1.PersistentVolumeMode(v1.PersistentVolumeFilesystem), 5),
+			Entry("with block single clone", v1.PersistentVolumeMode(v1.PersistentVolumeBlock), 1),
+			Entry("with block multiple clones", v1.PersistentVolumeMode(v1.PersistentVolumeBlock), 5),
```

Comment:
wonder how these multiple tests are working with the shared PVC?

Reply:
Can we be more explicit about when doing efficient snapshot clone vs dumb clone?

Reply:
The func tests are only doing efficient right now:
https://github.com/kubevirt/containerized-data-importer/blob/e62c96ffe0a76655466828fbfba8e646a672e381/tests/cloner_test.go#L2562

The reason I didn't do dumb was because it was hard to simulate a real env for the fallback to happen,
but I think I can give it a try now that the sourceVolumeMode stuff is in

Reply:
Apparently beta for this feature is 1.27
https://github.com/kubernetes/enhancements/issues/3141

So unless we enable stuff explicitly we won't have sourceVolumeMode populated for now

Reply:
I have something more hacky in mind, but at least we won't be tied to "local". Going to use AllowVolumeExpansion
Working on this right now

Reply:
Added host assisted tests

> wonder how these multiple tests are working with the shared PVC?

I guess because the temp source PVC has some k8s finalizers that protect it from being removed completely?